model/model.go

```diff
@@ -317,12 +340,23 @@ func build(kubeClient *kubernetes.Clientset, routerDeployment *v1beta1ext.Deploy
 	return routerConfig, nil
 }
 
-func buildRouterConfig(routerDeployment *v1beta1.Deployment, platformCertSecret *v1.Secret, dhParamSecret *v1.Secret) (*RouterConfig, error) {
+func buildRouterConfig(routerDeployment *v1beta1.Deployment, configMap *v1.ConfigMap, platformCertSecret *v1.Secret, dhParamSecret *v1.Secret) (*RouterConfig, error) {
 	routerConfig := newRouterConfig()
-	err := modeler.MapToModel(routerDeployment.Annotations, "nginx", routerConfig)
+
+	if configMap.Data != nil {
+		err := configmapModeler.MapToModel(configMap.Data, "nginx", routerConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// To maintain backwards compatibility with previous Deis releases and Kube < 1.2
+	// annotations on the router Deployment override the ConfigMap.
+	err := annotationModeler.MapToModel(routerDeployment.Annotations, "nginx", routerConfig)
```

Comment:
Two things here:

First, we really need to verify that these two sources of configuration can be merged cleanly. Consider (as one for instance) that a `RouterConfig` instance has a nested `GzipConfig` (or `SSLConfig`). If the second source of configuration (annotations) override _one_ attribute of one of those nested objects, we need to verify that it doesn't blank out all the _other_ attributes of those nested objects. (This is a _classic_ problem in merging maps of maps.) I _think_ it works ok in this particular case, but we really should double check.

The second thing (which might obviate the first)... do we really want to merge the two sources of configuration? I don't like the "config spray" where config values affecting the model could be found in so many different places. It might be simpler from an operator's perspective if this were as simple as "configuration comes from the configMap if it exists, else it comes from annotations; no merging of the two takes place."